modules/actors/actor-sheet-ffg.js

```diff
@@ -654,6 +742,83 @@ export class ActorSheetFFG extends ActorSheet {
         await DiceHelpers.rollSkill(this, event, upgradeType);
       });
 
+    // Roll crew
+    html.find(".roll-button-crew").children().on("click", async (event) => {
+      let crew_id = $(event.currentTarget).parents(".item").data("itemId");
+      let roles = crew_id.split('-'); // ship_id, crew_id, crew_role
+      // look up the sheet for passing to the roller
+      let crew_member = game.actors.get(roles[1]);
+      if (crew_member === undefined) {
+        ui.notifications.warn(game.i18n.localize("SWFFG.Crew.Removed"));
+        deregister_crew(roles[0], roles[1], roles[2]);
+        return;
+      }
+      let that = game.actors.get(roles[1]).sheet;
+      let starting_pool = {'difficulty': 2};
+      let handling = game.actors.get(roles[0]).system.stats.handling.value;
+
+      // add modifiers from the vehicle handling
+      if (handling > 0) {
+        starting_pool['boost'] = handling;
+      } else if (handling < 0) {
+        starting_pool['setback'] = handling * -1;
+      }
+
+      if (roles[2] === 'Pilot (Space)') {
+        let pool = new DicePoolFFG(starting_pool);
+        // look up the actor dice pool
+        pool = get_dice_pool(roles[1], 'Piloting: Space', pool);
```

Comment:
I think it does need some level of refactor (based on the comment from @Rysarian that it doesn't work for Genesys) but I don't agree with passing it in blindly. The role name is localized before being displayed to the user and localizing the stored role means it would not work anymore if the language of the system is changed.

Reply:
You're essentially doing something 3 times in this class, and each time you're calling the same method, so better to handle the `if (roles[2] === 'whatever')` stuff once, in a single location.

That way this class with its 3 different if statements just becomes a single call to `get_dice_pool` and doesn't have to care what the values of the `roles` elements are. Makes it easier to add or remove potential values later on too.

Reply:
(fine for a later refactor though)